allensdk/experimental/lazy_property.py

```diff
@@ -8,6 +8,9 @@ def __get__(self, obj, objtype=None):
         if obj is None:
             return self
 
+        if not hasattr(obj.api, self.getter_name):
```

Comment:
If I understand correctly condition will be met if a user builds a session object with an API that does not support access to some type of data, then requests those data anyways. This sounds more like an error than a None (analogous to a KeyError when accessing data not stored in a dictionary).

This could be pretty dangerous as bool(None) is False. Imagine that 
1. there is some data attribute which returns as a list
2. a user iterates through sessions, using the emptiness of that list to govern behavior e.g. `if thing_that_ought_to_be_a_list: do something else do something else`

Reply:
how about:
```
raise NotImplentedError(f"API class {self.api.__class__.__name__} does not support {self.getter_name}")
```